File: utils/js/fileReader.js
const fs = require('fs');

class FileReader {
    constructor(filePath) {
        this.filePath = filePath;
    }

    readLines() {
        try {
            const data = fs.readFileSync(this.filePath, 'utf8');
            return data.split('\n').map(line => line.trim());
        } catch (err) {
            console.error(`Error reading file ${this.filePath}:`, err.message);
            return [];
        }
    }
}

module.exports = FileReader;

File: 2024/utils/python/FileReader.py
class FileReader:
    def __init__(self, file_path):
        self.file_path = file_path

    def read_lines(self):
        try:
            with open(self.file_path, 'r') as file:
                lines = [line.strip() for line in file.readlines()]
            return lines
        except FileNotFoundError:
            print(f"Error: The file '{self.file_path}' was not found.")
            return []
        except IOError:
            print(f"Error: An error occurred while reading the file '{self.file_path}'.")
            return []
